File: examples/C/fip.c
/*
 *
 * Copyright (c) 2011, 2012
 *   University of Houston System and Oak Ridge National Laboratory.
 * 
 * All rights reserved.
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 
 * o Redistributions of source code must retain the above copyright notice,
 *   this list of conditions and the following disclaimer.
 * 
 * o Redistributions in binary form must reproduce the above copyright
 *   notice, this list of conditions and the following disclaimer in the
 *   documentation and/or other materials provided with the distribution.
 * 
 * o Neither the name of the University of Houston System, Oak Ridge
 *   National Laboratory nor the names of its contributors may be used to
 *   endorse or promote products derived from this software without specific
 *   prior written permission.
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
 * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
 * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
 * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
 * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
 * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
 * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
 * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
 * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *
 */



/*
 * float value put to shmalloc'ed variable
 *
 */

#include <stdio.h>
#include <math.h>

#include <shmem.h>

static const float e = 2.71828182;

static const float epsilon = 0.00000001;

int
main (void)
{
  float *f;
  int me;

  start_pes (0);
  me = _my_pe ();

  f = (float *) shmalloc (sizeof (*f));

  *f = 3.1415927;
  shmem_barrier_all ();

  if (me == 0)
    {
      shmem_float_p (f, e, 1);
    }

  shmem_barrier_all ();

  if (me == 1)
    {
      printf ("%s\n", (fabsf (*f - e) < epsilon) ? "OK" : "FAIL");
    }

  shfree (f);

  return 0;
}

File: examples/Fortran/fip.f90
! (c) 2011 University of Houston System.  All rights reserved.


!
! float value put to shmalloc'ed variable
!
program fip
  real e
  parameter ( e = 2.71828182 )
  real epsilon
  parameter ( epsilon = 0.00000001 )
  real, save :: f
  integer me

  include 'shmem.fh'

  call start_pes(0)
  me = my_pe()

  f = 3.1415927

  call shmem_barrier_all()

  if (me == 0) then
     call shmem_real_put(f, e, 1, 1)
  end if

 call shmem_barrier_all()
  
  if (me == 1) then
     if (abs(f - e) < epsilon) then
        print *, 'OK'
     else
        print *, 'FAIL'
     end if
  end if

end program fip
